### ISSUE TITLE
Add tests for overloads changed in .Net

### ISSUE THREAD

**Issue by S-dn-Y** [janestreet/PS7CompatibilityRules#1]
Hi!

Great initiative, I can see this get added into our pipeline!
It needs some more tests though..
With the .NET changes that come with PowerShell 7 some overloads where changed/added as well, which may impact PowerShell scripts that utilize them.

For example:
This will split in PowerShell 5:
```
PS C:\> [string]$PSVersionTable['PSVersion']
5.1.20348.2849
PS C:\> 'SomeExampleExampleExample'.Split('ex')
Som
E
ampl
E
ampl
E
ampl
```

But not in PowerShell 7 cause a different overload is hit:
```
PS C:\> [string]$PSVersionTable['PSVersion']
7.4.5
PS C:\> 'SomeExampleExampleExample'.Split('ex')
SomeExampleExampleExample
PS C:\>
```

While I was working on migrating our code base I've tried to search for a big list of these changes but wasn't quite successful. Split is one I know changed, but there may be others.

Best regards,
Sidney

**Comment by Tadas** [comment]
Hi,

Thanks for raising an issue!

The biggest problem with trying to implement a generic rule that could catch such overload confusion is simply knowing the type that the method is being called on. This was a lesson learned when creating the [NoHtmlParsing](https://github.com/janestreet/PS7CompatibilityRules/blob/main/NoHtmlParsing.psm1), you end up having to make some assumptions while trying to avoid false-positives.

Off the top of my head, some method invocation examples:
- `$var.Method()`
- `(expresison).Method()`
- `$hashtable['item'].Method()`
- `$arrayItem[0].Method()`
- `$wholeArray.Method()`

All of them would look differently in AST. Once you've done searching for such bits of AST you still need to walk back one "level" to try and figure out the type that the method is being called on. It's just not practical because PowerShell is dynamically typed.

With that said, it's probably more realistic to implement a rule that specifically looks for problematic invocations of `.Split()` and other commonly used .NET methods. I have this on my TODO, will try to dedicate some time to it.


Sligtly related, when https://github.com/PowerShell/PowerShell/pull/21320 hopefully lands it will be a useful tool to troubleshoot such overload confusion issues.

**Comment by Tadas** [comment]
Pushed https://github.com/janestreet/PS7CompatibilityRules/commit/a9442630115be048174aebb8c56ceabcf56f8f06 that adds a rule to warn about problematic `.Split()` calls. It should be relatively easy to expand with other methods when they come up.
